### PR TITLE
Remove torch distribution wrappers and rename parameters

### PR DESCRIPTION
**Pull request by neerajprad**
Addresses distribution renaming from #827:
 - Removes all distribution wrappers.
 - Renames distribution parameters in tests, examples and tutorials.
 - Also re-enables smoke test on `gp.ipynb` but disables `pyro.validation` (see #953).

This was done using a combination of automated replacements, and manual vetting. #961

### CODE REVIEW

**Comment by neerajprad** [comment]
Ready for review. This passes all our tests locally.

File: tutorial/source/bayesian_regression.ipynb [hunk 1]
```diff
@@ -80,7 +80,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Note that we generate the data with a fixed observation noise $\\sigma = 0.1$.\n",
+    "Note that we generate the data with a fixed observation noise $\\scale = 0.1$.\n",
```
Comment (fritzo) [inline tutorial/source/bayesian_regression.ipynb]: This use of `\sigma` is actually greek, so I think we want to keep it. I expect this PR will result in a latex error.

File: tutorial/source/gp.ipynb [hunk 2]
```diff
@@ -19,8 +19,8 @@
     "$$y = f + \\epsilon,\\quad \\epsilon \\sim \\mathcal{N}\\left(0, \\beta^{-1}\\mathbf{I}\\right).$$\n",
     "\n",
     "We will use [radial basis function kernel](https://en.wikipedia.org/wiki/Radial_basis_function_kernel) (RBF kernel) as the kernel of Gaussian process:\n",
-    "$$ k(x,x') = \\sigma^2 \\exp\\left(-\\frac{\\|x-x'\\|^2}{2l^2}\\right),$$\n",
-    "where $\\sigma^2$ is variance and $l$ is lengthscale of the RBF kernel."
+    "$$ k(x,x') = \\scale^2 \\exp\\left(-\\frac{\\|x-x'\\|^2}{2l^2}\\right),$$\n",
```
Comment (fritzo) [inline tutorial/source/gp.ipynb]: ditto: mind the latex greeks

File: tutorial/source/gp.ipynb [hunk 3]
```diff
@@ -58,8 +58,8 @@
     "pyro.set_rng_seed(0)\n",
     "\n",
     "import os\n",
-    "smoke_test = False # Flaky MVN constraint checking issue - see:https://github.com/uber/pyro/issues/953 \n",
-    "pyro.enable_validation(True)"
+    "smoke_test = ('CI' in os.environ)\n",
```
Comment (fritzo) [inline tutorial/source/gp.ipynb]: :+1:

File: tutorial/source/bayesian_regression.ipynb (hunk 1 above)
Comment (neerajprad) [reply tutorial/source/bayesian_regression.ipynb]: Oops..didn't realize the latex incantations here. Will revert these.

**Comment by neerajprad** [comment]
Ready to merge.

**Pull request merged**